src/query/sql/src/planner/binder/merge_into.rs

```diff
@@ -144,7 +144,7 @@ impl Binder {
             let mut name_map = HashMap::<String, Vec<IndexType>>::new();
             for column in source_output_columns {
                 name_map
-                    .entry(column.column_name.clone())
+                    .entry(column.column_name.as_str().trim_matches('`').to_string())
```

Comment:
We should remove '`' from `column_name` when binding columns, `subquery` will also report errors due to this problem
```sql
mysql> select a from (select a from test);
+--------------------+
| a                  |
+--------------------+
| 0.0941881516339449 |
| 0.8833429252655464 |
+--------------------+
2 rows in set (0.05 sec)
Read 2 rows, 17.00 B in 0.007 sec., 306.9 rows/sec., 2.55 KiB/sec.

mysql> select a from (select `a` from test);
ERROR 1105 (HY000): SemanticError. Code: 1065, Text = error:
  --> SQL:1:8
  |
1 | select a from (select `a` from test)
  |        ^ column a doesn't exist

mysql> select `a` from (select `a` from test);
ERROR 1105 (HY000): SemanticError. Code: 1065, Text = error:
  --> SQL:1:8
  |
1 | select `a` from (select `a` from test)
  |        ^^^ column a doesn't exist, do you mean 'a'?
```